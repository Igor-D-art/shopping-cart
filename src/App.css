.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 100vw;
}

.navigation{
  display: flex;
  justify-content: space-around;
  align-items: center;
  position: relative;
  width: 100%;
  height: 110px;
  background-color: black;
}

.cart-img {
  width: 30px;
  height: auto;
}

.nav-links{
  display: flex;
  align-items: center;
}

.navigation a{
  margin: 5px 30px 5px 30px;
  color: white;
  font-size: 24px;
  text-decoration: none;
}

.home, .contacts, .shop, .cart{
  position: relative;
  display: flex;
  height: 100%;
}

.shop{
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  height: fit-content;
  gap: 10px;
}

.shopCard{
  display: flex;
  flex-direction: column;
  border: 1px solid black;
  width: 200px;
  height: 220px;
  top: 110px;
}

.shopCardImage{
  position: relative;
  width: 50%;
  height: 50%;
}

.shopCardImage>img{
  position: relative;
  width: 100%;
  height: 100%;
}

.shopCardDetails{

}