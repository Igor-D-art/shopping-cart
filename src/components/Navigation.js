import React, { useContext } from "react";
import { Link } from "react-router-dom";
import CartContext from "../context/CartContext";

function Navigation(props) {
    
    let cart = useContext(CartContext);

    if (!cart ) {
        cart = []
    }
    
    const totalQty = () => {
        let sum = 0;
        for (let i = 0; i < cart.length; i++){
            sum += cart[i].qty 
        }
        return sum;
    }
    
    return (
        <nav className="navigation" data-testid='header'>
            <Link to={'/'}><h1 className="shop-name">BARAHLISHKO</h1></Link>
            <div className="link-container">
                <ul className="nav-links">
                    <div className="link-wrapper">
                        <Link to={'/'}> Home </Link>
                    </div>
                    
                    <div className="link-wrapper">
                        <Link to={'/shop'}> Shop </Link>
                    </div>

                    <div className="link-wrapper">
                        <Link to={'/contacts'}> Contacts </Link>
                    </div>

                    <div className="link-wrapper">

                        <Link to={'/cart'}>
                            {(cart.length===0) ? (
                                <img className="cart-img" src='images/cart.png' alt="" />
                            ) : (
                                <div className="nav-cart">
                                    <img className="cart-img" src='images/cart.png' alt="" />
                                    <p className="cart-qty">{ totalQty() }</p>
                                </div>
                            )}
                            
                        </Link>
                    </div>
                </ul>
            </div>
        </nav>
    )

}

export default Navigation;