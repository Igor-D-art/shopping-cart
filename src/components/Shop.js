import React, { useContext } from "react";
import ShopCard from "./ShopCard";
import ProductsContext from "../context/ProductsContext";


function Shop(props) {

    const products = useContext(ProductsContext);
    console.log(products)

    return (
        <div className="shop">
            {products.length? (products.map((product) => {
                const key = products.indexOf(product);
                return (
                       <ShopCard key={key} 
                        addToCart={props.addToCart}
                        product={product} />

                )   
                })) : (<h3> Fetching your data. Please wait a few milliseconds....</h3>)
            }
        </div>
    )
}

export default Shop;