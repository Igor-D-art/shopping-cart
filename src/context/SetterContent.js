import React from "react";

const SetterContext = React.createContext();

export const SetterProvider = SetterContext.Provider;


export default SetterContext;