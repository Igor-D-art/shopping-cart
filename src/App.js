import './App.css';
import React from 'react';
import Home from './components/Home';
import Navigation from './components/Navigation';
import Shop from './components/Shop';
import Contacts from './components/Contacts';
import Cart from './components/Cart';
import { BrowserRouter, Routes, Route } from "react-router-dom";


function App() {
  return (
    <BrowserRouter>
      <div className="App">
        {/* nav component renders everywhere */}
        <Navigation/>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/shop" element={<Shop />} />
          <Route path="/shop/:id" element={<Shop />} />
          <Route path="/contacts" element={<Contacts />} />
          <Route path="/cart" element={<Cart/>}/>
        </Routes>
      </div>
    </BrowserRouter>
  );
}

export default App;
