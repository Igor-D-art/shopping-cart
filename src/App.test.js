import { render, screen } from "@testing-library/react";
import App from "./App";
import Shop from "./components/Shop";
import '@testing-library/jest-dom';
import {ProductsProvider} from "./context/ProductsContext";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import Navigation from "./components/Navigation";


describe("General navigation check", () => {

  const navigation = <BrowserRouter><Navigation/></BrowserRouter>

  it('Navigation header is present on every page', () => {
    render(<App />);
    const navigation = screen.getByTestId('header');
    expect(navigation).toBeInTheDocument();
  })

  it("Navigation header contains shop name", () => {
    render(navigation)
    const heading = screen.getByRole('heading')
    expect(heading.textContent).toMatch('BARAHLISHKO');
  });

  it('Navigation header contains 5 links', () => {
    render(navigation);
    const links = screen.getAllByRole('link');
    expect(links.length).toBe(5);
  })

  it('Navigation header matches snapshot (contains links to Home, shop, contacts, cart', () => {
    render(navigation)
    const header = screen.getByTestId('header');
    expect(header).toMatchSnapshot();
  });

  it('User lands on a shop page when clicks the shop link', () => {
    // some code will go here
  })

});


describe('Testing shop', () => {

  it('shop contains productss', () => {

    const products = [{
      id: 1,
      title: "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
      price: 109.95,
      description: "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
      category: "men's clothing",
      image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
      rating: {
        rate: 3.9,
        count: 120,
      }
    }]

    render(
      <BrowserRouter>
        <ProductsProvider value={products}>
          <Shop/>
        </ProductsProvider>
      </BrowserRouter>
    )

    expect(screen.getByText("Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops")).toBeInTheDocument();
  });

});
