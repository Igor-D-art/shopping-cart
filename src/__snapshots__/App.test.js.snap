// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General navigation check Navigation header matches snapshot (contains links to Home, shop, contacts, cart 1`] = `
<nav
  class="navigation"
  data-testid="header"
>
  <a
    href="/"
  >
    <h1
      class="shop-name"
    >
      BARAHLISHKO
    </h1>
  </a>
  <div
    class="link-container"
  >
    <ul
      class="nav-links"
    >
      <div
        class="link-wrapper"
      >
        <a
          href="/"
        >
           Home 
        </a>
      </div>
      <div
        class="link-wrapper"
      >
        <a
          href="/shop"
        >
           Shop 
        </a>
      </div>
      <div
        class="link-wrapper"
      >
        <a
          href="/contacts"
        >
           Contacts 
        </a>
      </div>
      <div
        class="link-wrapper"
      >
        <a
          href="/cart"
        >
          <img
            alt=""
            class="cart-img"
            src="images/cart.png"
          />
        </a>
      </div>
    </ul>
  </div>
</nav>
`;
